daemon/api.go

```diff
@@ -1539,16 +1539,25 @@ func getSnapConf(c *Command, r *http.Request, user *auth.UserState) Response {
 	snapName := vars["name"]
 
 	keys := splitQS(r.URL.Query().Get("keys"))
-	if len(keys) == 0 {
-		return BadRequest("cannot obtain configuration: no keys supplied")
-	}
 
 	s := c.d.overlord.State()
 	s.Lock()
 	tr := config.NewTransaction(s)
 	s.Unlock()
 
 	currentConfValues := make(map[string]interface{})
+	// Special case - return root document
+	if len(keys) == 0 {
```

Comment:
Ok, done with a slight change to the logic you proposed because if root document is requested I don't want it to be buried under "" key in currentConfValues, so it's just retrued directly via SyncResponse.